Source/Renderer/ComputeEngine.js

```diff
@@ -0,0 +1,194 @@
+                    })
+                },
+                // Workaround Internet Explorer 11.0.8 lack of TRIANGLE_FAN
+                indices : new Uint16Array([0, 1, 2, 0, 2, 3]),
+                primitiveType : PrimitiveType.TRIANGLES
+            });
+
+            vertexArray = VertexArray.fromGeometry({
+                context : computeEngine._context,
+                geometry : geometry,
+                attributeLocations : {
+                    position : 0,
+                    textureCoordinates : 1
+                },
+                bufferUsage : BufferUsage.STATIC_DRAW,
+                interleave : true
+            });
+
+            computeEngine._viewportQuadVertexArray = vertexArray;
+        }
+
+        return vertexArray;
+    }
+
+    var viewportQuadAttributeLocations = {
+        position : 0,
+        textureCoordinates : 1
+    };
+
+    function createFramebuffer(context, texture) {
+        var fbo = new Framebuffer({
+            context : context,
+            colorTextures : [texture]
+        });
+        fbo.destroyAttachments = false;
+        return fbo;
+    }
+
+    function createViewportQuadShader(context, fragmentShaderSource) {
+        return ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : ViewportQuadVS,
+            fragmentShaderSource : fragmentShaderSource,
+            attributeLocations : viewportQuadAttributeLocations
+        });
+    }
+
+    function createRenderState(width, height)
+    {
+        return RenderState.fromCache({
+            viewport : new BoundingRectangle(0.0, 0.0, width, height)
+        });
+    }
+
+    ComputeEngine.prototype.execute = function(computeCommand) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(computeCommand)) {
+            throw new DeveloperError('computeCommand is required.');
+        }
+
+        if (!defined(computeCommand.fragmentShaderSource) && !defined(computeCommand.shaderProgram)) {
+            throw new DeveloperError('computeCommand.fragmentShaderSource or computeCommand.shaderProgram is required.');
+        }
+
+        if(!defined(computeCommand.outputTexture)) {
+            throw new DeveloperError('computeCommand.outputTexture is required.');
+        }
+        //>>includeEnd('debug');
+
+        var texture = computeCommand.outputTexture;
+        var width = texture.width;
+        var height = texture.height;
+
+        var context = this._context;
+        var vertexArray = computeCommand.vertexArray || createViewportQuadVertexArray(this);
+        var shaderProgram = computeCommand.shaderProgram || createViewportQuadShader(context, computeCommand.fragmentShaderSource);
+        var framebuffer = createFramebuffer(context, texture);
+        var renderState = createRenderState(width, height);
+        var uniformMap = defaultValue(computeCommand.uniformMap, defaultValue.EMPTY_OBJECT);
+
+        var clearCommand = new ClearCommand({
```

Comment:
Avoid allocating this and `DrawCommand` below by using scratch variables.  For example, see [here](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Cartesian3.js#L322).